uaclient/entitlements/tests/test_fips.py

```diff
@@ -35,49 +38,78 @@
+        self.entitlement = FIPSEntitlement(self.cfg)
+
+
+class TestFIPSEntitlementCanEnable(BaseEnabledFIPSConfig):
+
+    @mock.patch('uaclient.util.is_container', mock.Mock(return_value=False))
     @mock.patch('os.getuid', return_value=0)
     def test_can_enable_true_on_entitlement_inactive(self, m_getuid):
         """When operational status is INACTIVE, can_enable returns True."""
-        tmp_dir = self.tmp_dir()
-        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
-        cfg.write_cache('machine-token', FIPS_MACHINE_TOKEN)
-        cfg.write_cache('machine-access-fips', FIPS_RESOURCE_ENTITLED)
-        entitlement = FIPSEntitlement(cfg)
-        # Unset static affordance container check
-        entitlement.static_affordances = ()
+        op_status, op_status_details = self.entitlement.operational_status()
+        assert status.INACTIVE == op_status
+        assert 'FIPS PPA is not configured' == op_status_details
+        with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
+            assert True is self.entitlement.can_enable()
+        assert '' == m_stdout.getvalue()
+
+    @mock.patch('uaclient.util.is_container', mock.Mock(return_value=True))
+    @mock.patch('os.getuid', return_value=0)
+    def test_can_enable_false_on_containers(self, m_getuid):
+        """On containers, can_enable is false and operational status n/a."""
+        op_status, op_status_details = self.entitlement.operational_status()
+        assert status.INAPPLICABLE == op_status
+        msg_cant_install = 'Cannot install FIPS on a container'
+        assert msg_cant_install == op_status_details
         with mock.patch('sys.stdout', new_callable=StringIO) as m_stdout:
-            self.assertTrue(entitlement.can_enable())
-        self.assertEqual('', m_stdout.getvalue())
+            assert False is self.entitlement.can_enable()
+        assert msg_cant_install + '\n' == m_stdout.getvalue()
 
 
-class TestFIPSEntitlementEnable(TestCase):
+class TestFIPSEntitlementEnable(BaseEnabledFIPSConfig):
 
-    @mock.patch('uaclient.util.get_platform_info')
-    @mock.patch('os.getuid', return_value=0)
-    def test_enable_configures_apt_sources_and_auth_files(
-            self, m_getuid, m_platform_info):
-        """When entitled, configure apt repo auth token, pinning and url."""
-        m_platform_info.return_value = 'xenial'
-        tmp_dir = self.tmp_dir()
-        cfg = config.UAConfig(cfg={'data_dir': tmp_dir})
-        cfg.write_cache('machine-token', FIPS_MACHINE_TOKEN)
-        cfg.write_cache('machine-access-fips', FIPS_RESOURCE_ENTITLED)
-        entitlement = FIPSEntitlement(cfg)
-        # Unset static affordance container check
-        entitlement.static_affordances = ()
-
-        with mock.patch('uaclient.apt.add_auth_apt_repo') as m_add_apt:
+    @mock.patch('uaclient.util.get_platform_info', return_value='xenial')
+    def test_enable_configures_apt_sources_and_auth_files_and_installs(
+            self, _platform_info):
```

Comment:
If we did `s/return_value='xenial'/Mock(\0)/` in the decorator, we wouldn't get the mock passed in at all.